paasta_tools/cli/cmds/spark_run.py

```diff
@@ -0,0 +1,384 @@
+import json
+import os
+import socket
+import sys
+from random import randint
+
+from botocore.session import Session
+
+from paasta_tools.cli.cmds.check import makefile_responds_to
+from paasta_tools.cli.cmds.cook_image import paasta_cook_image
+from paasta_tools.cli.utils import check_if_port_free
+from paasta_tools.cli.utils import get_instance_config
+from paasta_tools.cli.utils import lazy_choices_completer
+from paasta_tools.cli.utils import list_instances
+from paasta_tools.cli.utils import list_services
+from paasta_tools.cli.utils import pick_random_port
+from paasta_tools.mesos_tools import find_mesos_leader
+from paasta_tools.mesos_tools import MESOS_MASTER_PORT
+from paasta_tools.utils import _run
+from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import get_username
+from paasta_tools.utils import load_system_paasta_config
+from paasta_tools.utils import NoConfigurationForServiceError
+from paasta_tools.utils import NoDeploymentsAvailable
+from paasta_tools.utils import NoDockerImageError
+from paasta_tools.utils import paasta_print
+from paasta_tools.utils import PaastaColors
+from paasta_tools.utils import PaastaNotConfiguredError
+from paasta_tools.utils import SystemPaastaConfig
+
+DEFAULT_SPARK_UI_PORT = 4040
+DEFAULT_SPARK_WORK_DIR = '/spark_client'
+DEFAULT_SPARK_DOCKER_IMAGE_PREFIX = 'paasta-spark-run'
+DEFAULT_SPARK_DOCKER_REGISTRY = 'docker-dev.yelpcorp.com'
+
+
+def add_subparser(subparsers):
+    list_parser = subparsers.add_parser(
+        'spark-run',
+        help="Run Spark on the PaaSTA cluster",
+        description=(
+            "'paasta spark-run' is useful for launching a Spark cluster on the PaaSTA mesos "
```

Comment:
Grammatical tidy-up:
```
'paasta spark-run'  launches a Spark cluster on PaaSTA. It analyzes soa-configs and command line arguments invoke construct a 'docker run'. By default, it will pull the Spark service image from the registry unless the --build option is used.
```